pkg/cloudprovider/providers/azure/azure_cache.go

```diff
@@ -19,63 +19,138 @@ package azure
 import (
 	"sync"
 	"time"
-
-	"k8s.io/client-go/tools/cache"
 )
 
-type timedcacheEntry struct {
-	key  string
-	data interface{}
-}
+type getFunc func(key string) (interface{}, error)
+type listFunc func() (map[string]interface{}, error)
+
+// timedCache is a cache with TTL.
+type timedCache struct {
+	store       map[string]interface{}
+	lock        sync.Mutex
+	ttl         time.Duration
```

Comment:
move ttl to per entry bases, expiry check should be on per key bases.

Reply:
see https://github.com/kubernetes/kubernetes/pull/59253#discussion_r165794935